src/Stratis.Bitcoin/Database/LevelDb.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System.Linq;
+using LevelDB;
+using NBitcoin;
+
+namespace Stratis.Bitcoin.Database
+{
+    /// <summary>A minimal LevelDb wrapper that makes it compliant with the <see cref="IDb"/> interface.</summary>
+    public class LevelDb : IDb
+    {
+        private string dbPath;
+
+        private DB db;
+
+        public IDbIterator GetIterator(byte table)
+        {
+            return new LevelDbIterator(table, this.db.CreateIterator());
+        }
+
+        public void Open(string dbPath)
+        {
+            this.dbPath = dbPath;
+            this.db = new DB(new Options() { CreateIfMissing = true }, dbPath);
+        }
+
+        public void Clear()
+        {
+            this.db.Dispose();
+            System.IO.Directory.Delete(this.dbPath, true);
+            this.db = new DB(new Options() { CreateIfMissing = true }, this.dbPath);
+        }
+
+        public IDbBatch GetWriteBatch() => new LevelDbBatch(this.db);
+
+        public byte[] Get(byte table, byte[] key)
+        {
+            return this.db.Get(new[] { table }.Concat(key).ToArray());
+        }
+
+        public void Dispose()
+        {
+            this.db.Dispose();
+        }
+    }
+
+    /// <summary>A minimal LevelDb wrapper that makes it compliant with the <see cref="IDbBatch"/> interface.</summary>
+    public class LevelDbBatch : WriteBatch, IDbBatch
+    {
+        private DB db;
+
+        public LevelDbBatch(DB db)
+        {
+            this.db = db;
+        }
+
+        public IDbBatch Put(byte table, byte[] key, byte[] value)
+        {
+            return (IDbBatch)this.Put(new[] { table }.Concat(key).ToArray(), value);
+        }
+
+        public IDbBatch Delete(byte table, byte[] key)
+        {
+            return (IDbBatch)this.Delete(new[] { table }.Concat(key).ToArray());
+        }
+
+        public void Write()
+        {
+            this.db.Write(this, new WriteOptions() { Sync = true });
+        }
+    }
+
+    /// <summary>A minimal LevelDb wrapper that makes it compliant with the <see cref="IDbIterator"/> interface.</summary>
+    public class LevelDbIterator : IDbIterator
+    {
+        private byte table;
+        private Iterator iterator;
+
+        public LevelDbIterator(byte table, Iterator iterator)
+        {
+            this.table = table;
+            this.iterator = iterator;
+        }
+
+        public void Seek(byte[] key)
+        {
+            this.iterator.Seek(new[] { this.table }.Concat(key).ToArray());
+        }
+
+        public void SeekToLast()
```

Comment:
Could you give a high-level description of how this method works? It is not entirely intuitive to me how seeking based on a single-byte table identifier gives the last record. Is it the last record within the current table? I presume it has something to do with how records are arranged below the table identifiers.